src/main/java/de/tum/in/www1/artemis/service/LectureService.java

```diff
@@ -133,6 +140,15 @@ public SearchResultPageDTO<Lecture> getAllOnPageWithSize(final SearchTermPageabl
      * @param lecture the lecture to be deleted
      */
     public void delete(Lecture lecture) {
+        if (pyrisWebhookService.isPresent()) {
+            Lecture lectureWithAttachmentUnits = lectureRepository.findByIdWithLectureUnitsAndAttachmentsElseThrow(lecture.getId());
+            List<AttachmentUnit> attachmentUnitList = lectureWithAttachmentUnits.getLectureUnits().stream().filter(lectureUnit -> lectureUnit.getType().equals("attachment"))
+                    .map(lectureUnit -> (AttachmentUnit) lectureUnit).collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Why use `Collectors.toCollection(ArrayList::new)` here? does the list have to be mutable?

Reply:
Collectors.toList() gives back an error in the Style tests, thus the use of Collectors.toCollection(ArrayList::new)